homeassistant/components/switch/rachio.py

```diff
@@ -0,0 +1,214 @@
+
+
+# Represents one Rachio Iro
+class RachioIro(object):
+    def __init__(self, r, device_id):
+        self.r = r
+        self._device_id = device_id
+        self._device = None
+        self._running = None
+        self._zones = None
+
+    def __str__(self):
+        return "Rachio Iro " + self.serial_number
+
+    @property
+    def device_id(self):
+        return self._device['id']
+
+    @property
+    def status(self):
+        return self._device['status']
+
+    @property
+    def serial_number(self):
+        return self._device['serialNumber']
+
+    @property
+    def is_paused(self):
+        return self._device['paused']
+
+    @property
+    def is_on(self):
+        return self._device['on']
+
+    @property
+    def current_schedule(self):
+        return self._running
+
+    def list_zones(self, include_disabled=False):
+        if not self._zones:
+            self._zones = [RachioZone(self.r, self, zone['id'])
+                           for zone in self._device['zones']]
+
+        if include_disabled:
+            return self._zones
+        else:
+            self.update(propagate=True, no_throttle=True)
+            return [z for z in self._zones if z.is_enabled]
+
+    # Pull updated device info from the Rachio API
+    @util.Throttle(MIN_UPDATE_INTERVAL, MIN_FORCED_UPDATE_INTERVAL)
+    def update(self, propagate=True):
+        self._device = self.r.device.get(self._device_id)[1]
+        self._running = self.r.device.getCurrentSchedule(self._device_id)[1]
+
+        # Possibly update all zones
+        if propagate:
+            for zone in self.list_zones(include_disabled=True):
+                zone.update(propagate=False)
+
+        _LOGGER.debug("Updated " + str(self))
+
+
+# Represents one zone of sprinklers connected to the Rachio Iro
+class RachioZone(SwitchDevice):
+    def __init__(self, r, device, zone_id):
+        self.r = r
+        self._device = device
+        self._zone_id = zone_id
+        self._zone = None
+
+    def __str__(self):
+        return "Rachio Zone " + self.name
+
+    @property
+    def zone_id(self):
+        return self._zone['id']
+
+    @property
+    def unique_id(self):
+        return '{iro}-{zone}'.format(
+            iro=self._device.device_id,
+            zone=self.zone_id)
+
+    @property
+    def number(self):
+        return self._zone['zoneNumber']
+
+    @property
+    def name(self):
+        return self._zone['name'] or "Zone " + self.number
+
+    @property
+    def is_enabled(self):
+        return self._zone['enabled']
+
+    # TODO: fix this always returning false
+    @property
+    def is_on(self):
+        self._device.update(propagate=False)
+        schedule = self._device.current_schedule
+        return self.zone_id == schedule.get('zoneId')
+
+    # Pull updated zone info from the Rachio API
+    def update(self, propagate=True):
+        self._zone = self.r.zone.get(self._zone_id)[1]
+
+        # Possibly update device
+        if propagate:
```

Comment:
Your entity is set to be polled for updates by Home Assistant (because you do not overwrite the `should_poll` property). Each entity will individually call `_device.update()`, which is fine because `Throttle` will take care of it only getting called once.

So you can remove the whole propagation of updates.